### PR TITLE
feat!: replace tier4_map_msgs with autoware_map_msgs for MapProjectorInfo

### PR DESCRIPTION
**Pull request by mitsudome-r**
## Description
This pull request replace tier4_map_msgs with autoware_map_msgs for MapProjectorInfo message.

## Related links

**Parent Issue:**

- https://github.com/autowarefoundation/autoware.core/issues/99

## How was this PR tested?

Tested locally with planning simulator.
(Also tested internally in TIER IV: https://evaluation.ci.tier4.jp/evaluation/reports/13a50fac-d48a-5d70-bb2d-bc162b72dffe?project_id=prd_jt, https://evaluation.ci.tier4.jp/evaluation/reports/da3a38df-8607-57ea-a588-0a6e6ef212d2?project_id=prd_jt)

## Notes for reviewers

None.

#### Modifications

| Version | Topic Type      | Topic Name        | Message Type        | Description       |
|:--------|:----------------|:------------------|:--------------------|:------------------|
| Old     | Pub/Sub| `/map/map_projector_info` | `tier4_map_msgs/MapProjectorInfo` | projection information for map frame |
| New     | Pub/Sub | `/map/map_projector_info` | `autoware_map_msgs/MapProjectorInfo` | projection information for map frame |

## Effects on system behavior

None.

### CODE REVIEW

**Review by isamu-takagi — APPROVED** [review]
LGTM (component_interface_specs)

**Comment by YamatoAndo** [comment]
@mitsudome-r please fix cppcheck-differential

**Comment by YamatoAndo** [comment]
@mitsudome-r fix this file
https://github.com/autowarefoundation/autoware.universe/blob/c369209f770a9b22e436c44847697e355e35ebb3/map/autoware_map_projection_loader/test/data/map_projector_info_local.yaml

File: map/autoware_map_projection_loader/src/map_projection_loader.cpp [hunk 1]
```diff
@@ -25,39 +25,50 @@
...
     msg.map_origin.altitude = data["map_origin"]["altitude"].as<double>();
 
-  } else if (msg.projector_type == tier4_map_msgs::msg::MapProjectorInfo::LOCAL) {
+  } else if (msg.projector_type == autoware_map_msgs::msg::MapProjectorInfo::LOCAL) {
     ;  // do nothing
 
+  } else if (msg.projector_type == "local") {
+    RCLCPP_WARN_STREAM(
+      rclcpp::get_logger("MapProjectionLoader"),
+      "Load " << filename << std::endl
```
Comment (YamatoAndo) [inline map/autoware_map_projection_loader/src/map_projection_loader.cpp:52]:
```suggestion
      "Load " << filename << "\n"
```

File: map/autoware_map_projection_loader/src/map_projection_loader.cpp [hunk 2]
```diff
@@ -25,39 +25,50 @@
...
+              << "DEPRECATED WARNING: projector type \"local\" is deprecated."
+                 "Please use \"Local\" instead. For more info, visit "
+                 "https://github.com/autowarefoundation/autoware.universe/blob/main/map/"
+                 "map_projection_loader/"
+                 "README.md"
+              << std::endl);
```
Comment (YamatoAndo) [inline map/autoware_map_projection_loader/src/map_projection_loader.cpp:58]:
```suggestion
              );
```

**Comment by YamatoAndo** [comment]
@mitsudome-r Please check carefully whether other repositories need to be modified.

https://github.com/autowarefoundation/autoware_lanelet2_extension/blob/36f5ffb7692ad218ce610d35256882ca38f37488/autoware_lanelet2_extension/docs/lanelet2_format_extension.md?plain=1#L117

**Comment by mitsudome-r** [comment]
@YamatoAndo Thank you.
I will create the PRs for other repositories as well. I will also update the sample map in the documentation to make sure that they are aligned.

File: map/autoware_map_projection_loader/src/map_projection_loader.cpp (hunk 1 above)
Comment (mitsudome-r) [reply map/autoware_map_projection_loader/src/map_projection_loader.cpp]: fixed in https://github.com/autowarefoundation/autoware.universe/pull/9392/commits/81625e3a7e39f45068b623e7b63207fe7b575554

File: map/autoware_map_projection_loader/src/map_projection_loader.cpp (hunk 2 above)
Comment (mitsudome-r) [reply map/autoware_map_projection_loader/src/map_projection_loader.cpp]: fixed in https://github.com/autowarefoundation/autoware.universe/pull/9392/commits/81625e3a7e39f45068b623e7b63207fe7b575554

**Review by YamatoAndo — APPROVED** [review]

**Review by kosuke55 — APPROVED** [review]

**Review by maxime-clem — APPROVED** [review]
OK for `autoware_carla_interface`.

**Pull request merged**